src/app/src/components/DashboardDeleteFacilityControls.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect } from 'react';
+import { arrayOf, bool, func, string } from 'prop-types';
+import Button from '@material-ui/core/Button';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import Typography from '@material-ui/core/Typography';
+import { toast } from 'react-toastify';
+import get from 'lodash/get';
+
+import { facilityDetailsPropType } from '../util/propTypes';
+
+const deleteControlsStyles = Object.freeze({
+    containerStyles: Object.freeze({
+        width: '100%',
+        padding: '10px 0',
+        margin: '10px 20px',
+        display: 'flex',
+        justifyContent: 'start',
+        alignItems: 'center',
+    }),
+    buttonStyles: Object.freeze({
+        margin: '0 20px 10px 0',
+    }),
+    labelStyles: Object.freeze({
+        fontSize: '16px',
+    }),
+});
+
+export default function DashboardDeleteFacilityControls({
+    handleDelete,
+    deleting,
+    error,
+    disabled,
+    data,
+}) {
+    const [dialogIsOpen, setDialogIsOpen] = useState(false);
+    const [deletingFacility, setDeletingFacility] = useState(false);
```

Comment:
Yeah, that's right -- this is here in part because there doesn't seem to be an easy way to diff `prevProps` and `props` while using the `useEffect` hook like there would be in `componentDidUpdate`. The local `deletingFacility` state is a stand-in for having `prevProps`. 

`deletingFacility` could also potentially live in Redux, but it sort of only exists so that this component knows to close the dialog and show the toast on successful requests -- both of which are things it controls directly.